src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+    // `isLiquid` dictates whether the position is liquid or not
+    // If true:
+    //      position can support user withdraws
+    // else:
+    //      position can not support user withdraws
+    //====================================================================
+
+    /**
+     * @notice Attempted to interact with a Convex market pid & baseRewardPool that the Cellar is not using.
+     */
+    error ConvexAdaptor__ConvexBoosterPositionsMustBeTracked(
+        uint256 pid,
+        address baseRewardPool,
+        ERC20 lpt,
+        CurvePool _curvePool,
+        bytes4 _selector
+    );
+
+    /**
+     * @notice Attempted to pass adaptorData that does not comply with the stored information within Convex Booster records.
+     */
+    error ConvexAdaptor__ConvexBoosterPositionsDoesNotMatchAdaptorData(
+        uint256 pid,
+        address baseRewardPool,
+        ERC20 lpt,
+        CurvePool pool,
+        bytes4 selector
+    );
+
+    /**
+     * @notice CurvePool is in a re-entered state.
+     */
+    error ConvexCurveAdaptor___PoolInReenteredState();
+
+    /**
+     * @param _booster the Convex booster contract for the network/market (different booster for Curve, FRAX, Prisma, etc.)
+     * @dev Booster.sol serves as the primary contract that accounts for markets via poolIds. PoolInfo structs can be queried w/ poolIds, where baseRewardPool contracts, and other info can be obtained.
+     */
+    constructor(address _booster) {
+        booster = IBooster(_booster);
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Convex Curve Adaptor V 0.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice Deposit & Stakes LPT from the cellar into Convex Market at the end of the user deposit sequence.
+     * @param assets amount of LPT to deposit and stake
+     * @param adaptorData see adaptorData info at top of this smart contract
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        (uint256 pid, address rewardsPool, ERC20 lpt, CurvePool pool, bytes4 selector) = abi.decode(
+            adaptorData,
+            (uint256, address, ERC20, CurvePool, bytes4)
+        );
+
+        if (selector != bytes4(0)) _callReentrancyFunction(pool, selector);
+        else revert BaseAdaptor__UserDepositsNotAllowed();
+
+        _validatePositionIsUsed(pid, rewardsPool, lpt, pool, selector);
```

Comment:
Can you move this line above the call reentrancy line?

Reply:
Done.